src/index.ts

```diff
@@ -365,11 +365,28 @@ const typescript: PluginImpl<RPT2Options> = (options) =>
 				}
 				else
 				{
-					const relativePath = relative(pluginOptions.cwd, fileName);
+					// don't mutate the entry because generateBundle gets called multiple times
+					let entryText = entry.text
```

Comment:
this took me a few hours to debug and was by far the most time-consuming piece 😕  I originally changed `entry.text` itself which then got changed on top of itself for each output...

Ideally I think `generateBundle` should have some caching so this isn't repeated per output. I could do that with a cache key inside of `declarations{}`, but there's a good question as to what that cache key should be (in this case, I think the `declarationDir` variable is the only thing that is subject to change,  but that sounds brittle if more stuff gets added).